ci/docker/runtime_functions.sh

```diff
@@ -313,16 +313,22 @@ build_ubuntu_amalgamation_min() {
 build_ubuntu_gpu_cmake_mkldnn() {
     set -ex
     cd /work/build
+    # We need to use generic archtecture. Otherwise, MKLDNN compiled in one
+    # CPU architecture (e.g., C5) can't run on another architecture (e.g., g3).
     cmake \
         -DUSE_CUDA=1               \
         -DUSE_CUDNN=1              \
         -DUSE_MKLML_MKL=1          \
         -DUSE_MKLDNN=1             \
         -DCMAKE_BUILD_TYPE=Release \
+        -DARCH_OPT_FLAGS="-mtune=generic" \
```

Comment:
This was never a problem before and is counter intuitive to the user. It should be possible to compile on one instance type and execute on another instance type without having to specify options like this one. I'd appreciate it if you could track this down.

Reply:
this problem is caused by how your CI system is set up. It's compiled on C5, which supports AVX512, and runs on g3, which doesn't support AVX512. MKLDNN by default tries to optimize specifically for an architecture where it is compiled. There is similar setup for Makefile.

Reply:
I see, so MKLDNN auto-assumes the runtime architecture based on the runtime-architecture. I think this will cause quite some problems for our users in production environments, considering everybody will build their binaries in a build fleet and deploy it in a production fleet with different configuration and hardware.

Could you please elaborate the flow how users are being made aware of this problem and how we instruct them to solve this problem? Also, where is ARCH_OPT_FLAGS being consumed?

Reply:
Would be good if @piiswrong @szha @cjolivier01 could look into the possible impact of that change

Reply:
the same thing has been used in prepare_mkldnn.sh, which is used by Makefile.

Reply:
https://github.com/apache/incubator-mxnet/blob/master/prepare_mkldnn.sh#L96

Reply:
The difference here is that the linked script is only for the compilation of MKLDNN - maybe @pengzhao-intel can elaborate. Here, you're changing the compilation of the entire MXNet library.

Reply:
If this is what you worry about, I have changed CMakeLists.txt to only compile MKLDNN with the flag.
https://github.com/apache/incubator-mxnet/pull/10731/files#diff-af3b638bc2a3e6c650974192a53c7291R162